pandas/core/missing.py

```diff
@@ -219,16 +219,38 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
     return new_y
 
 
-def interpolate_2d(values, method='pad', axis=0, limit=None, fill_value=None, dtype=None):
-    """ perform an actual interpolation of values, values will be make 2-d if
-    needed fills inplace, returns the result
+def pad(values, method='pad', axis=0, limit=None, fill_value=None, dtype=None):
+    """ 
+    Perform an actual interpolation of values. 1-d values will be made 2-d temporarily. 
+    Returns the result
     """
 
+    ndim = values.ndim
+
+    if ndim > 2:
+        slice_axis = 1 if axis == 0 else 0
+        slicer = [slice(None), slice(None)]
```

Comment:
Is there a reason to include axes >1? The slice axis will always be either 0 or 1, so the other slices are always implied to be full slices. Is it a matter of clarity?